server/src/main/java/org/opensearch/search/aggregations/bucket/histogram/AutoDateHistogramAggregator.java

```diff
@@ -288,13 +296,132 @@ private static class FromSingle extends AutoDateHistogramAggregator {
+                        // Ensure that the query and aggregation are on the same field
+                        if (prq.getField().equals(fieldName)) {
+                            createFilterForAggregations(fieldName,
+                                NumericUtils.sortableBytesToLong(prq.getLowerPoint(), 0),
+                                NumericUtils.sortableBytesToLong(prq.getUpperPoint(), 0));
+                        }
+                    }
+                } else if (context.query() instanceof ConstantScoreQuery) {
+                    final ConstantScoreQuery csq = (ConstantScoreQuery) context.query();
+                    // Ensure that the constant score query is instance of match all query
+                    if (csq.getQuery() instanceof MatchAllDocsQuery) {
+                        findBoundsAndCreateFilters(fieldName, context);
+                    }
+                } else if (context.query() instanceof MatchAllDocsQuery) {
+                    findBoundsAndCreateFilters(fieldName, context);
+                }
+            }
+        }
+
+        private void findBoundsAndCreateFilters(final String fieldName, final SearchContext context) throws IOException {
+            final List<LeafReaderContext> leaves = context.searcher().getIndexReader().leaves();
+            long min = Long.MAX_VALUE, max = Long.MIN_VALUE;
+            // Since the query does not specify bounds for aggregation, we can
+            // build the global min/max from local min/max within each segment
+            for (LeafReaderContext leaf : leaves) {
+                min = Math.min(min, NumericUtils.sortableBytesToLong(
+                    leaf.reader().getPointValues(fieldName).getMinPackedValue(), 0));
+                max = Math.max(max, NumericUtils.sortableBytesToLong(
+                    leaf.reader().getPointValues(fieldName).getMaxPackedValue(), 0));
+            }
+            createFilterForAggregations(fieldName, min, max);
+        }
+
+        private void createFilterForAggregations(final String field, final long low, final long high) throws IOException {
+            // max - min / targetBuckets = bestDuration
+            // find the right innerInterval this bestDuration belongs to
+            // since we cannot exceed targetBuckets, bestDuration should go up,
+            // so the right innerInterval should be an upper bound
+            long bestDuration = (high - low) / targetBuckets;
+            long interval = 0;
+            roundingInfosLoop: do {
+                RoundingInfo curRoundingInfo = roundingInfos[roundingIdx];
+                for (int curInnerInterval: curRoundingInfo.innerIntervals) {
+                    if (bestDuration <= curInnerInterval * curRoundingInfo.roughEstimateDurationMillis) {
+                        interval = curInnerInterval * curRoundingInfo.roughEstimateDurationMillis;
+                        break roundingInfosLoop;
+                    }
+                }
+                roundingIdx++;
```

Comment:
Probably need to move the preparedRounding as well to the next index? Else it is always staying at 0 (second level)